pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -239,7 +239,13 @@ func (r *ReconcilePravegaCluster) deploySegmentStore(p *pravegav1alpha1.PravegaC
 		return err
 	}
 
-	configMap := pravega.MakeSegmentstoreConfigMap(p)
+	cm := &corev1.ConfigMap{}
+	name := p.Spec.Pravega.SegmentStoreConfigMap
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, cm)
```

Comment:
This code is not needed. Please add directly to podspec as stated above. There is no need to confirm if the config map exists. If it does not deployment will fail and that is the expected behavior.